File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int calculate_words(string text);
int calculate_letters(string text);
int calculate_sentences(string text);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int words = calculate_words(text);
    int letters = calculate_letters(text);
    int sentences = calculate_sentences(text);

    // Compute the Coleman-Liau index
    double L = (double) letters / words * 100;
    double S = (double) sentences / words * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Print the grade level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int calculate_words(string text)
{
    int words = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            words += 1;
        }
    }
    return words;
}

int calculate_letters(string text)
{
    int letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters += 1;
        }
    }
    return letters;
}

int calculate_sentences(string text)
{
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences += 1;
        }
    }
    return sentences;
}

File: sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")

    words_count = calculate_words(text)
    letters_count = calculate_letters(text)
    sentences_count = calculate_sentences(text)

    index = get_index(words_count, letters_count, sentences_count)

    print_grade(index)


def calculate_words(text):
    words = 1
    for word in text:
        if word == " ":
            words += 1
    return words


def calculate_letters(text):
    letters = 0
    for word in text:
        if word.isalpha():
            letters += 1
    return letters


def calculate_sentences(text):
    sentences = 0
    for word in text:
        if word == "." or word == "?" or word == "!":
            sentences += 1
    return sentences


def get_index(words, letters, sentences):
    l = letters / words * 100
    s = sentences / words * 100
    return round(0.0588 * l - 0.296 * s - 15.8)


def print_grade(index):
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


main()
